core/Common.php

```diff
@@ -1024,13 +1024,13 @@ public static function getContinent($country)
      *            1 => array( ... ) // campaign keyword parameters
      * );
      */
-    public static function getCampaignParameters(?int $idSite = null)
+    public static function getCampaignParameters(?int $idSite = null, bool $checkSetting = true)
     {
         $featureFlagManager = StaticContainer::get(FeatureFlagManager::class);
         if ($featureFlagManager->isFeatureActive(PrivacyCompliance::class)) {
             $cache = TrackerCache::getCacheWebsiteAttributes($idSite);
             $cacheKey = CampaignTrackingParametersDisabled::class;
-            if (($cache[$cacheKey] ?? false) === true) {
+            if (($cache[$cacheKey] ?? false) === true && $checkSetting) {
                 return [[], []];
             }
         }
```

Comment:
Might be good to move the check out so we don't even instantiate the other classes when not needed.

```suggestion
    if ($checkSetting) {
        $featureFlagManager = StaticContainer::get(FeatureFlagManager::class);
        if ($featureFlagManager->isFeatureActive(PrivacyCompliance::class)) {
            $cache = TrackerCache::getCacheWebsiteAttributes($idSite);
            $cacheKey = CampaignTrackingParametersDisabled::class;
            if (($cache[$cacheKey] ?? false) === true) {
                return [[], []];
        }
    }
```

Reply:
Done.